### PR TITLE
Some general cleanup, edits, and expanded hello_world/installation pages.

### PR DESCRIPTION
**Pull request by eX-n0x**
### About this PR:
Attempts to freshen up the document as well as adding critical information and cleaning linting errors that could cause issues with parsers.

### Why is this needed:
_bababooey_

### Changelog:
- added installation guide for byond
- added explanation on how to use dream maker in hello_world
- fixed numerous linting issues
- fixed some links
- various grammar edits
- slightly modified some page styles
- other stuff maybe

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -4,40 +4,40 @@
 [Installation](./installation.md)
 
 - [Hello World](./hello_world.md)
-	- [Comments](./hello/comments.md)
-	- [Basic Types](./hello/types.md)
```
Comment (ZeWaka) [inline src/SUMMARY.md]: hard tabs, not spaces.

File: src/hello/comments.md [hunk 2]
```diff
@@ -3,12 +3,12 @@
 Any program requires comments, and DM supports a few different varieties:
 
 * Regular comments which are ignored by the compiler:
-	* `// Line comments which go to the end of the line.`
-	* `/* Block comments which go to the closing delimiter. */`
```
Comment (ZeWaka) [inline src/hello/comments.md]: hard tabs, not spaces.

File: src/preprocessor.md [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
 
-Sometimes, you'll want to have shared values across your project, instead of copy-pasting them everywhere. This is where `#define` comes in.
+Sometimes, you'll want to have shared values across your project instead of copy-pasting them everywhere. This is where `#define` comes in.
 
-The `#define Name Value` statement substitues the `Name` for `Value` for as long as the define is not `#undef`ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were defined. That is different from `"This is [BIG]."`, where `BIG` is an embedded expression, which does get processed for macro substitution.
+The `#define Name Value` statement substitues the `Name` for `Value` as long as the define is not `#undef`-ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were to be defined. That's different from `"This is [BIG]."`, whereas `BIG` is an embedded expression, which does get processed for macro substitution.
```
Comment (ZeWaka) [inline src/preprocessor.md]:
```suggestion
The `#define Name Value` statement substitues the `Name` for `Value` as long for as the define is not `#undef`-ined. Substitution only applies to whole words. Text inside of double or single quotes is not processed for substitution, so `"This is BIG."` would not be modified even if a define named `BIG` were to be defined. That's different from `"This is [BIG]."`, whereas `BIG` is an embedded expression, which does get processed for macro substitution.
```
the `for` is intentional.

File: src/ops/bitflags.md [hunk 4]
```diff
@@ -27,5 +27,6 @@ Since DM only supports floating point numbers, we are restricted to 24 bits (23
 For a closer look at the binary math behind this, check out tgstation's [article](https://tgstation13.org/wiki/Binary_flags).
 
 ## Activity
-Using the above code snippet as a base, try implementing the ability to toggle a flag easily.
-> Hint: What other binary operators are there?
+
+>Using the above code snippet as a base, try implementing the ability to toggle a flag easily.
+>>**Hint**: What other binary operators are there?
```
Comment (ZeWaka) [inline src/ops/bitflags.md:32]: is this valid syntax? GitHub doesn't think so.

File: src/objs/disposal.md [hunk 5]
```diff
@@ -15,19 +17,23 @@ Del(foo)
 
 The garbage collector works by using a reference counting system. Once an object is no longer referenced, it gets deleted.
 
-Important to note: circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
+>**Important to Note**: Circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
```
Comment (ZeWaka) [inline src/objs/disposal.md]:
```suggestion
>**Important to note**: Circular references will never be deleted by the garbage collector. This is defined as a pair of objects with variables that point to each other, or even an object with a variable that points to itself. Also, an object with running or sleeping procs will not be deleted.
```
why capitalize

File: src/hello_world.md [hunk 6]
```diff
@@ -14,15 +16,23 @@ This is the source code of the traditional Hello World program.
 
 `world << "x"` is the most basic way to output text to the default window, using the << [operator].
 
-To run this, simply open up a blank byond project, and compile it.
+## Starting Dream Maker
```
Comment (ZeWaka) [inline src/hello_world.md]: I would prefer the guide not use Dream Maker unless necessary - see sprites.

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: src/ops/bitflags.md (hunk 4 above)
Comment (eX-n0x) [reply src/ops/bitflags.md:32]: Should be, both the VSCode preview and GitHub style preview 
Screenshot 2021-06-16 182926
render it without issues.

**Pull request merged**